js-pkg/apps/demos/src/components/webrtc.js

```diff
@@ -0,0 +1,149 @@
+import * as React from 'react'
+import { useDriftDBSignalingChannel } from './signaling'
+
+export const useWebRTCMessagingChannel = (p1, p2) => {
+  let theirdataChannelRef = React.useRef(null)
+  let [messages, addMessage] = React.useReducer((state, msg) => [...state, msg], [])
+  let [latency, setLatency] = React.useState(0)
+  const connSetupArray = React.useRef([
+    (conn) => {
+      let dataChannel = conn.createDataChannel(p1)
+      dataChannel.onmessage = (e) => {
+        addMessage({ id: e.timeStamp, text: e.data })
+      }
+
+      conn.ondatachannel = (e) => {
+        if (e.channel.label === p2) {
+          theirdataChannelRef.current = e.channel
+        }
+      }
+    }
+  ])
+  const getLatency = useWebRTCConnection(p1, p2, connSetupArray.current)
+  React.useEffect(() => {
+    const latency_setter = setTimeout(async function l() {
+      setLatency(await getLatency())
+      setTimeout(l, 1000)
```

Comment:
You'll need to assign this to `latency_setter`. (Maybe `timeoutId` is a better name btw?)

Reply:
aaaaa, always forget that. thank you.